metrology/registry.py

```diff
@@ -53,19 +53,22 @@ def derive(self, name):
 
     def get(self, name):
         with self.lock:
-            return self.metrics[name]
+            key = self._compose_key(name)
+            return self.metrics[key]
 
     def add(self, name, metric):
```

Comment:
Why not adding an extra parameters for tags?

```suggestion
    def add(self, name, metric, tags = []):
```

Reply:
That's what I did in the other branch I mentioned in the pull request. I didn't like how it's dragging through the whole project and leads to some suboptimal parameter orders for the function signatures. But if you like this style better, I can pull request this other branch :)
Although it needs some additional work.